tools/ossl-version.pl, util/mktar.sh: New version extraction script, use it

levitte opened the pull request:
util/mktar.sh was still adapted to pre-3.0 OpenSSL version data.

tools/ossl-version.pl is a new script that extracts OpenSSL version
data from any version of opensslv.h and outputs a set of shell
variable definitions reflecting the version data:

    OPENSSL_VERSION_STR         just the numbers
    OPENSSL_FULL_VERSION_STR    numbers plus possible pre-release tag
    OPENSSL_VERSION_COMPAT      the numbers this release is upward
                                compatible with

We adapt util/mktar.sh to use tools/ossl-version.pl to figure out the
version part of the tarball name.

We also install tools/ossl-version.pl for the benefit of scripts that
want to use it as well.  The usage differs depending on the operating
system family.  For Unix shell and MSWindows command lines, it outputs
appropriate variable setting lines, to be used as the user sees fit.
For VMS, it additionally defines environment variables with the same
name, which may end up as CLI symbols or logical names, depending on
the definition of the logical name PERL_ENV_TABLES.

Thanks to Roumen Petrov for notifying us

levitte committed:
tools/ossl-version.pl, util/mktar.sh: New version extraction script, use it

util/mktar.sh was still adapted to pre-3.0 OpenSSL version data.

tools/ossl-version.pl is a new script that extracts OpenSSL version
data from any version of opensslv.h and outputs a set of shell
variable definitions reflecting the version data:

    OPENSSL_VERSION_STR         just the numbers
    OPENSSL_FULL_VERSION_STR    numbers plus possible pre-release tag
    OPENSSL_VERSION_COMPAT      the numbers this release is upward
                                compatible with

We adapt util/mktar.sh to use tools/ossl-version.pl to figure out the
version part of the tarball name.

We also install tools/ossl-version.pl for the benefit of scripts that
want to use it as well.  The usage differs depending on the operating
system family.  For Unix shell and MSWindows command lines, it outputs
appropriate variable setting lines, to be used as the user sees fit.
For VMS, it additionally defines environment variables with the same
name, which may end up as CLI symbols or logical names, depending on
the definition of the logical name PERL_ENV_TABLES.

Thanks to Roumen Petrov for notifying us

levitte commented:
I'm considering renaming util/version.pl to tools/ossl-version.pl and ensure it's installed, so that external scripts can use it as well.

mspncp commented:
Note: There is a ‘u’ missing in the first name: It’s Roumen Petrov (or  Румен Петров).

levitte committed:
Configure, Configurations/common.tmpl: Handle scripts from source

The configuration code assumed that all scripts are generated from
some kind of source.  However, some scripts may be ready to use and
install directly from the source, and this needs to be detected.

levitte commented:
> Note: There is a ‘u’ missing in the first name: It’s Roumen Petrov (or Румен Петров).

Ah, thanks.  Will fix

levitte commented:
This includes an additional commit to properly detect scripts that can be installed directly from source

mspncp commented:
This one seems to be forgotten in ready state?

levitte commented:
I recall that @petrovr said something about this, er, *somewhere*, but can't remember where to look.  Something was not entirely right.  @petrovr, would you mind reminding me, here?

(this is why you should discuss a PR in that PR's discussion stream and not somewhere else.  It's hard enough keeping track of all the discussions already)

levitte commented:
Considering #10205, I don't think this is useful any more.  Does everyone agree?

mspncp commented:
> Does everyone agree?

**ENOTIMETOTHINKABOUTIT**

_(I did not follow the conversation close enough, just happened notice that the pr was stalled in the ready state.)_

mattcaswell commented:
I agree

beldmit commented:
> **ENOTIMETOTHINKABOUTIT**

!!!

mspncp commented:
> Considering #10205, I don't think this is useful any more. Does everyone agree?

Everybody agreed, so let's close it.
